### PR TITLE
Allow the raw Geometry to be passed to methods

### LINKED ISSUE #15: GeometryProxy needs to extend GeoPHP's Geometry
FatalThrowableError in Geometry.class.php line 273:
Type error: Argument 1 passed to Geometry::contains() must be an instance of Geometry, array given

```
$polygon = \App\Polygon::find(10);
    $point = Geo::parseWkt("POINT(46.7193603515625 24.780890141913442)");
    $geo =Geo::parseGeoJson($polygon->geojson);
     $result=$geo->contains($point);
    dd($result);
```

### PR DESCRIPTION
**Pull request by jimmypuckett**
This allows the geometries to interact with each other.  Will fix #15 .

### CODE REVIEW

**Comment by jimmypuckett** [comment]
@ssfinney This is just a WIP, so don't accept it yet.  I am not 100% sold on the solution & I have not written any test yet.  I just pushed this up to start a discussion on the solution.

**Comment by jimmypuckett** [comment]
@ssfinney If this passes, then we are good to review.  I should have coverage back up to 100%.

File: src/Support/GeometryProxy.php [hunk 1]
```diff
@@ -163,11 +163,7 @@ private function determineCoordinateIndices($index, $length)
      */
     protected function exposeRawIfAvailable($argument)
     {
-        if (method_exists($argument, 'getRawGeometry')) {
-            return $argument->getRawGeometry();
-        }
-
-        return $argument;
+        return (method_exists($argument, 'getRawGeometry')) ? $argument->getRawGeometry() : $argument;
```
Comment (ssfinney) [inline src/Support/GeometryProxy.php]: Why make this into a ternary? This is less readable to me than splitting it up.

**Review by ssfinney — CHANGES_REQUESTED** [review]

File: src/Support/GeometryProxy.php (hunk 1 above)
Comment (jimmypuckett) [reply src/Support/GeometryProxy.php]: Well, I have an answer, but it is not a great one...

The code coverage sees it as one line & there is not a good way to write the test to get the else with the way that the method_exist works.  This gives me 100% without having to tweak my test ;-)

I don't mind putting it back the other way, but I am not going to mess with he required changes to the test.

Just let me know what you prefer...

Comment (ssfinney) [reply src/Support/GeometryProxy.php]: I would rather have cleaner code than an extra 1% coverage, so keep the old.

I understand why you would push for 100 though, it's addictive!

**Review by ssfinney — APPROVED** [review]

**Pull request merged**